site/content/en/docs/Tutorials/ebpf_tools_in_minikube.md

```diff
@@ -0,0 +1,64 @@
+---
+title: "Running eBPF Tools in Minikube"
+linkTitle: "Running eBPF Tools in Minikube"
+weight: 1
+date: 2019-08-15
+description: >
+  Running eBPF Tools in Minikube
+---
+
+## Overview
+
+eBPF tools are performance tools used for observing the Linux kernel.
+These tools can be used to monitor your Kubernetes application in minikube.
+This tutorial will cover how to set up your minikube cluster so that you can run eBPF tools from a Docker container within minikube. 
+
+## Prerequisites
+
+- minikube source code
+- Latest minikube binary
+
+## Tutorial
+
+First, you will need to build the minikube ISO from scratch.
+Instructions for building the ISO can be found [here](../Contributing/iso.md).
+
+_Note: Building the ISO can take at least an hour._
+
+Now, start minikube:
+
+```
+$ minikube start
+```
+
+Mount kernel headers that were built during the ISO build into the minikube VM:
+
+```
+$ cd minikube
+$ minikube mount $(pwd)/out/buildroot/output/build/linux-4.19.94:/lib/modules/4.19.94/build
```

Comment:
Hey @afbjorklund -- these kernel headers are required for ebpf tools to run, and they exist on my host filesystem when I build the ISO. I can't figure out how to include the headers in the ISO so that this step can be skipped -- do you have any guidance around this?

Reply:
For Falco we shipped a prebuilt kernel module, rather than providing users with compilers and headers at runtime

Reply:
One alternative approach: we could just fetch the appropriate kernel source tarball and extract the headers

Reply:
That's much better, I created a tarball of the headers which an be downloaded from GCS and updated the doc.

@afbjorklund I don't really know anything about prebuilding kernel modules, would you know of any good references to get started with that?